### PR TITLE
Migrated UpdateExperimentName endpoint from hibernate to jdbi Impl

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java [hunk 1]
```diff
@@ -483,4 +476,25 @@ public FutureExperimentDAO(
...
+            executor)
+        .thenCompose(
+            unused -> {
+              var name = request.getName();
+              if (request.getName().isEmpty()) {
+                name = MetadataServiceImpl.createRandomName();
+              }
+
+              return updateExperimentField(
+                  request.getId(), "name", ModelDBUtils.checkEntityNameLength(name));
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java:495]: Names cannot be duplicated. We're not checking that.

Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java]: Names cannot be duplicated. We're not checking that.

**Review by conradoverta — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java]: We are not checking duplication for Experiment entity. We were not checking duplication on old code as well. reference: https://github.com/VertaAI/modeldb/blob/daf936ab83230ea4565fd3f3e047e1d83a1b03b5/backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java#L230

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java]: Added check for experiment name already exists in project or not

File: backend/src/test/java/ai/verta/modeldb/ExperimentTest.java [hunk 2]
```diff
@@ -1865,4 +1865,42 @@ public void deleteExperiments() throws InterruptedException {
...
+          UpdateExperimentName.newBuilder()
+              .setId(experiment.getId())
+              .setName(experiment2.getName())
+              .build();
+
+      experimentServiceStub.updateExperimentName(updateNameRequest);
+      fail();
+    } catch (StatusRuntimeException ex) {
+      Status status = Status.fromThrowable(ex);
+      LOGGER.warn("Error Code : " + status.getCode() + " Description : " + status.getDescription());
```
Comment (liam-verta) [inline backend/src/test/java/ai/verta/modeldb/ExperimentTest.java]: This error is expected as part of the test. Info at most. 
```suggestion
      LOGGER.info("Error Code : " + status.getCode() + " Description : " + status.getDescription());
```

**Review by liam-verta — APPROVED** [review]

**Pull request merged**